packages/design-system/src/components/icon.tsx

```diff
@@ -1,38 +1,55 @@
 import { type VariantProps, cva } from 'class-variance-authority';
-import React from 'react';
-import type { IconType } from '../types/icon-types';
-import * as Icons from './icons';
+import { twMerge } from 'tailwind-merge';
 
-export const iconVariants = {
+type SvgIconProps = {
+  title?: string;
+} & React.SVGProps<SVGSVGElement>;
+
+type SvgIcon = React.FunctionComponent<SvgIconProps>;
+
+type BaseIconProps = {
+  icon: string | SvgIcon;
+} & VariantProps<typeof IconStyles> &
+  React.SVGProps<SVGSVGElement>;
+
+type ConditionalIconProps =
+  | {
+      title: string;
+      decorative: false;
+    }
+  | { title?: string; decorative: true };
+
+type IconProps = BaseIconProps & ConditionalIconProps;
+
+const IconStyles = cva('', {
   variants: {
     size: {
       small: 'ko:size-4',
       medium: 'ko:size-6',
       large: 'ko:size-8',
-      extraLarge: 'ko:size-10',
-      extraHuge: 'ko:size-14',
     },
   },
-} as const;
-
-const iconStyles = cva('', {
-  variants: iconVariants.variants,
   defaultVariants: {
     size: 'medium',
   },
 });
-
-type IconVariants = VariantProps<typeof iconStyles>;
-type IconSize = IconVariants['size'];
-export const iconSizes = Object.keys(iconVariants.variants.size) as IconSize[];
-
-type IconProps = {
-  name: IconType;
-  title: string;
-} & VariantProps<typeof iconStyles>;
-
-export function Icon({ name, size, title, ...props }: IconProps) {
-  const className = iconStyles({ size });
-  const IconComponent = React.createElement(Icons[name], { ...props, title, className });
-  return IconComponent;
+export function Icon({ icon, size, title, decorative, className, ...props }: IconProps) {
+  if (typeof icon === 'string') {
+    return (
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        aria-hidden={decorative ? 'true' : 'false'}
+        focusable="false"
+        role="img"
+        className={twMerge(IconStyles({ size }), className)}
+        viewBox="0 0 24 24"
+        fill="currentColor"
+      >
+        {title && <title>{title}</title>}
```

Comment:
Some screen readers cannot read the title tag and requires explicitly setting `aria-labelledby`:

```
<svg role="img" aria-labelledby="title">
  <title id="title">{title}</title>
</svg>
```

We should probably do that. The ID has to be unique so it should be generated.